### PR TITLE
GraphQL Removal, Bug Fixes, & Improvements

### LINKED ISSUE #408: fix ui
upload modal
list - shouldn't send haystack request. (run objects don't currently have a site name)

### PR DESCRIPTION
**Pull request by axelstudios**
# Improvements
- Completely removed GraphQL, all associated files, and dependencies. The UI is 100% on the REST API
- Migrated to AWS SDK v3 as v2 is being deprecated
- Migrated to React functional components with named exports
- Added ErrorDialog to view simulation error log in UI
- Fixed bad paths in svg logo, resized for Redoc usage
- Replaced `XMLHttpRequest` with fetch (via ky)
- Removed unused weatherFile handling from upload process
- The API is only instantiated once now
- Disabled upload button until file with valid extension is selected
- Aliases are now deleted with associated sites
- On the Sites page, removed Points lookup every second when the dialog isn't opened
- OSM & FMU sites can be selected simultaneously now, and all actions apply to both
- Sorted Points by name in PointDialog
- Removed `onInvokeAction` from alfalfa-server, REST API should be used instead
- Added hover text to simulation-completed timestamp with unformatted data (showing seconds, tz, etc)
- Removed unused dependencies
- Removed `Remove Test Results` button from Simulations page since it was unimplemented and undefined
- Refactored `ResultsDialog` and `FileInput` into their own files
- Disabled the StartDialog Run button when the end time predated the start time (or endSeconds predated startSeconds for fmu sims)
- Added messages to dialogs if no points/simulation results are available
- Added close buttons to dialogs
- Compressed all API responses, including those under 1kB
- Removed unused styles
- Replaced downloadjs with native S3 content-disposition header
- Updated dependencies & resolved vulnerabilities
- And many more

# Bug Fixes
- Fixed uploads
- Fixed broken FMU runs via UI (StartDialog inputs were still using seconds)
- Fixed docs generation
- Fixed bug where `listModels` wasn't returning the id
- Fixed deleting runs & points associated with deleted sites (`site` reference doesn't exist on runs any more)
- Fixed deleting orphaned siteRef redis key
- Fixed downloading simulations with `S3_URL_EXTERNAL`
- Fixed the StartDialog from disabling the time inputs when external or realtime were selected
- Fixed race condition corner case where loading points could fail
- Fixed `writeInputPoint` bugs
- Fixed broken `mc policy` usage
- Fixed potential `createModel` race condition
- When multiple sites are selected clicking a button (start/stop/remove) only acts on sites in the correct state (avoids 400 errors)
- Replaced deprecated `onBackdropClick` usage
- Removed deprecated @mui/styles
- Replaced deprecated `collection.insert` call
- Migrated to MUI X v6 to resolve console warnings
- Added error handling for invalid json input to prevent 500 error

# API Updates
- Added thorough documentation to every endpoint
- Added API documentation grouping by endpoint
- Added data types to generic inputs
- Added error handling to every endpoint
- Added input validation to endpoints with input
  - Verification that IDs exist
  - Type/uuid/regex/min/max verification of keys and values
  - OpenStudio/FMU simulation types verify different timestamp formats
- Added multiple endpoints (listing simulations, deleting aliases, etc)
- All timestamps returned by the API are now formatted to ISO-8601 UTC
- Updated Haystack documentation with links to the official docs for each op
- Removed non-api `/upload-url` endpoint

Closes #408 #388 #358 #276 #140

### CODE REVIEW

File: alfalfa_web/server/api-v2.js [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-const apiv2 = ({ db: _db, redis: _redis }) => {
-  db = _db;
-  redis = _redis;
-  api = new AlfalfaAPI(db, redis);
+const apiv2 = ({ api: _api }) => {
+  api = _api;
   return router;
 };
 
+const validate = (data, rules) => {
```
Comment (TShapinsky) [inline alfalfa_web/server/api-v2.js:24]: Very nice, I like it

**Review by TShapinsky — COMMENTED** [review]
This looks great Alex. I'm going to run some UI tests locally an then I'll approve if you're ready

**Comment by axelstudios** [comment]
@TShapinsky I included one last improvement, just for you:
image

**Comment by axelstudios** [comment]
A couple screenshots from the update:

##### API Documentation showing the new nav on the left and the level of detail in schemas
API Documentation

##### Error Dialog
ErrorDialog

##### Start Dialog (multiple changes)
image

**Review by TShapinsky — APPROVED** [review]
AWESOME!

**Pull request merged**